thanosql/resources/_table.py

```diff
@@ -142,43 +153,64 @@ def create(
     def upload(
         self,
         name: str,
-        file: Union[str, os.PathLike],
+        file: Optional[Union[str, os.PathLike]] = None,
+        df: Optional[pd.DataFrame] = None,
         schema: Optional[str] = None,
         table: Optional[TableObject] = None,
```

Comment:
Is 'table' an empty table schema? (parameter naming issue..)
And both are table objects, but there's no difference between a table schema and an actual table with records, right?

and

There is no code to prevent simultaneous use of df and file.

Reply:
Oh I will add the code to prevent simultaneous use of df and file.

'table' is a TableObject (columns and constraints). 'Table' is for 'BaseTable' (name, schema, columns, constraints) plus TableService (which allows records).. a bit confusing ㅠㅠ

Reply:
빠른 시일 내에 SDK에서 사용하는 변수, 오브젝트 등의 네이밍 및 통합 정리 작업을 해야겠습니다.